app/models/interactive_run_state.rb

```diff
@@ -72,24 +83,59 @@ def portal_hash
     end
   end
 
+  def parsed_interactive_state
+    JSON.parse(raw_data).symbolize_keys rescue {}
+  end
+
   def report_service_hash
+    # Expected data format can be checked in lara-typescript/interactive-api-client/metadata-types.ts:
+    # IRuntime<...>Metadata interfaces.
+    # Metadata is simply provided as a part of interactive state. It's optional and everything should work if there's
+    # no metadata defined or interactive state is empty.
+    metadata = parsed_interactive_state
+
+    # This property is defined in IRuntimeMetadataBase:
+    type = metadata[:answerType] || (interactive.has_report_url ? "external_link" : "interactive_state")
+
     result = {
+      # type can be overwritten by metadata[:answerType] prop (e.g. to "open_response_answer").
+      # Otherwise, the default "interactive_state" or "external_link" will be used.
+      type: type,
+      # question_type should match answer type.
+      question_type: InteractiveRunState.question_type_for(type),
+      # These properties are defined in IRuntimeMetadataBase:
+      submitted: metadata[:submitted],
+      answer_text: metadata[:answerText],
+      # These properties are stored in LARA. They're basic interactive run state properties. Some of them might be
+      # unused by Report or Portal when interactive pretends to be a basic question type. But these services might be
+      # extended to show both basic question answer and optionally provide iframe report view.
       id: answer_id,
-      question_type: "iframe_interactive",
-      question_id: interactive.embeddable_id
+      question_id: interactive.embeddable_id,
```

Comment:
I was worried about id conflicts with legacy question types. But I see that `embeddable_id` includes the embeddable class name with the id. So for these interactives it will either be `mw_interactive_[id]` or `managed_interactive_[id]`. 

And the answer_id does the same thing. So these ids are independent of the type or question_type.